src/Avalonia.Controls/TextBox.cs

```diff
@@ -446,6 +458,40 @@ public bool CanPaste
             private set { SetAndRaise(CanPasteProperty, ref _canPaste, value); }
         }
 
+        /// <summary>
+        /// Property for determining whether undo/redo is enabled
+        /// </summary>
+        public bool IsUndoEnabled
+        {
+            get { return GetValue(IsUndoEnabledProperty); }
+            set 
+            { 
+                SetValue(IsUndoEnabledProperty, value);
+                if (value == false)
```

Comment:
Well, I guess I can duplicate my comment here:

Property changed handling should be done in OnPropertyChanged method override or related observers. This code won't code if user will set property with SetValue(IsUndoEnabledProperty, value) directly.
For UndoLimit it's not that important, it's direct property that always will call setter, but makes sense for consistency I would say.